### PR TITLE
Upgrade JSPM & bundle

### PR DESCRIPTION
**Pull request by davidrapson**
An attempt to address https://github.com/guardian/developers-site/issues/64

> Our JSPM setup is lagging behind. Latest jspm should also allow us to easily bundle the app for much faster load, rather than relying on the CDN.

- Updates dependencies to use installed versions, not CDN
- Adds build task to gulp file

Probably a few things I've missed before this could be merged but feels close enough to get a second opinion. @theefer @OliverJAsh

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -30,24 +31,24 @@ gulp.task('css', function () {
         .pipe(gulp.dest('./target/css'));
 });
 
+gulp.task('js', function(done) {
+    jspm.bundleSFX('src/js/app', './target/js/bundle.js', {
```
Comment (davidrapson) [inline gulpfile.js]: Running the build via the API here as that seemed the nicest way to integrate with gulp. Not sure if it's preferable to just call out to `jspm bundle-sfx` as a shell task.

File: setup.sh [hunk 2]
```diff
@@ -20,8 +14,8 @@ bundle install --gemfile $BASE/Gemfile-sass
 
 npm install
 
-jspm dl-loader
+# Update jspm dl-loader
+./node_modules/jspm/jspm.js dl-loader
```
Comment (OliverJAsh) [inline setup.sh]: FYI you can also do `./node_modules/.bin/jspm`

Comment (davidrapson) [inline setup.sh]: Good call, thanks.

File: src/footer.ejs [hunk 3]
```diff
@@ -10,14 +10,7 @@
...
         </footer>
-        <script async>
-            /* global System */
-            System.import('js/app').catch(function(e) {
-                setTimeout(function() {
-                    throw e;
-                });
-            });
-        </script>
+        <script src="./js/bundle.js" async defer></script>
```
Comment (OliverJAsh) [inline src/footer.ejs]: We don't need to include `system.js` or `config.js` anymore if we're loading a SFX bundle?

Alternatively, we still include system+config and just do `jspm bundle --inject` (https://github.com/jspm/jspm-cli/blob/master/docs/production-workflows.md#loading-a-bundle-automatically-inject)

File: src/head.ejs [hunk 4]
```diff
@@ -24,17 +24,10 @@
...
 
-        <!-- TODO: Concatenate -->
-        <script src="./jspm_packages/system.js"></script>
-        <script src="./js/config.js"></script>
-        <script>
-            /* global System */
-            // Must be loaded before the body is parsed?
-            System.import('js/modernizr');
-        </script>
-        <script src="./js/picturefill.min.js" async></script>
```
Comment (OliverJAsh) [inline src/head.ejs]: Ahh, should've read this first—you have removed system+config. This means we lose the nice "refresh to reload" workflow in development. Could we keep this workflow whilst still bundling in production? (i.e. `jspm bundle --inject` https://github.com/jspm/jspm-cli/blob/master/docs/production-workflows.md#loading-a-bundle-automatically-inject)

Comment (davidrapson) [inline src/head.ejs]: Ah yeah, that's a really good point, don't want to loose that. I'll take another look at `bundle --inject`.

**Pull request closed without merging**